segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -118,14 +119,24 @@
     private final AtomicInteger readAtOffset = new AtomicInteger();
     private final AtomicInteger bytesToRead = new AtomicInteger();
     private final AtomicInteger currentArgIndex = new AtomicInteger();
+    private final AtomicLong currentIndexOffset = new AtomicLong();
 
-    DefragmentOperation(ChunkedSegmentStorage chunkedSegmentStorage, MetadataTransaction txn, SegmentMetadata segmentMetadata, String startChunkName, String lastChunkName, ArrayList<String> chunksToDelete) {
+    DefragmentOperation(ChunkedSegmentStorage chunkedSegmentStorage,
+                        MetadataTransaction txn,
+                        SegmentMetadata segmentMetadata,
+                        String startChunkName,
+                        String lastChunkName,
+                        ArrayList<String> chunksToDelete,
```

Comment:
Why ArrayList here and List below? Use the least restrictive type please.

Reply:
fixed.